plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java

```diff
@@ -601,13 +612,13 @@ public void onStaleStateChange(boolean isStale) {
     @Override
     public void onMapChanged(int change) {
       if (change == MapView.WILL_START_LOADING_MAP) {
-        onStop();
+        onLocationLayerStop();
       } else if (change == MapView.DID_FINISH_LOADING_STYLE) {
         locationLayer.initializeComponents(options);
         locationLayerCamera.initializeOptions(options);
         setRenderMode(locationLayer.getRenderMode());
```

Comment:
If the plugin is disabled and we change the style this line will draw the puck in the last location even though the plugin is disabled.

Reply:
Yes, want to make this clear that this is a bug that is in master though, not something I introduced in this PR...

Reply:
After looking into this, I'm not finding a solution to this fix. Will open a ticket and revisit in a separate PR.

Reply:
https://github.com/mapbox/mapbox-plugins-android/issues/515